hack/release/trigger-release-gate-pipelines.sh

```diff
@@ -0,0 +1,86 @@
+#! /bin/bash
+
+# Copyright 2021 VMware Tanzu Community Edition contributors. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+BOLD='\033[1m'
+CLEAR='\033[0m'
+
+function info {
+  BLUE='\033[0;34m'
+  local message=${1}
+
+  echo -e "${BLUE}${BOLD}${message}${CLEAR}"
+}
+
+function warning {
+  YELLOW='\033[0;33m'
+  local message=${1}
+
+  echo -e "${YELLOW}${BOLD}${message}${CLEAR}"
+}
+
+function error {
+  RED='\033[0;31m'
+  local message=${1}
+
+  echo -e "${RED}${BOLD}${message}${CLEAR}"
+}
+
+function success {
+  GREEN='\033[0;32m'
+  local message=${1}
+
+  echo -e "${GREEN}${BOLD}${message}${CLEAR}"
+}
+
+function check_dependencies {
+  # this script requires git & the gh CLI tool
+  warning "--- Checking dependencies ---"
+
+  if ! gh --version > /dev/null 2>&1; then
+    error "gh binary test failed. GitHub CLI tool required"
+    exit 1
+  fi
+
+  if ! gh auth status > /dev/null 2>&1; then
+    error "gh auth check failed. Ensure you've logged into to the gh CLI using \`gh auth login\` or have GITHUB_TOKEN set in your environment"
+    exit 1
+  fi
+
+  if [[ -z "${BUILD_VERSION:-}" ]]; then
+    error "BUILD_VERSION environment variable is not set"
+    exit 1
+  fi
+
+  success "--- Dependencies verified --- \n"
+}
+
+check_dependencies
+
+warning "Checking if ${BUILD_VERSION} release exists"
+
+if ! gh release view --repo https://github.com/vmware-tanzu/community-edition "${BUILD_VERSION}" > /dev/null 2>&1; then
+    error "${BUILD_VERSION} release does not exist. Ensure ${BUILD_VERSION} TCE release has been created"
+    exit 1
+fi
+
+info "Triggering MacOS and Linux build tests"
+
+# MacOS and Linux Build Test
+gh workflow run --repo https://github.com/vmware-tanzu/community-edition/ \
+  --ref main \
+  release-gate-macos-linux-builds.yaml \
+  --raw-field release_version="${BUILD_VERSION}"
+
+info "Triggering Windows build tests"
+
+# Windows Build Test
+gh workflow run --repo https://github.com/vmware-tanzu/community-edition/ \
```

Comment:
Just to confirm, is there some auth protection on this to make sure only certain people can trigger a release workflow?

Reply:
Agreed. This is how we do this when we cut a release:
https://github.com/vmware-tanzu/community-edition/blob/main/hack/release/cut-release.sh#L36-L40

This should be done at the top of the script.

Reply:
> Just to confirm, is there some auth protection on this to make sure only certain people can trigger a release workflow?

@stmcginnis I believe it's the default settings on the GitHub repo and the GitHub API. Not everyone can trigger GitHub Actions workflows. The authn/authz completely depends on the account being used with `gh`. In this case, we use `GITHUB_TOKEN` environment variable or ask the user to use `gh auth login` to authenticate - https://github.com/vmware-tanzu/community-edition/pull/2570/files#diff-dedc80ea59f623bfdbbaddf70b6c30e6fc18a7cc3fcadf7dd840c8346cee65caR50-R53

Reply:
> Agreed. This is how we do this when we cut a release: https://github.com/vmware-tanzu/community-edition/blob/main/hack/release/cut-release.sh#L36-L40
> 
> This should be done at the top of the script.

Ah yeah, I did see this, along with `# IMPORTANT: This should only ever be called CI/github-action` in `Makefile`.

I think the script can be used by devs too, it's okay and for the same reason I was a bit skeptical to put inside the `# IMPORTANT: This should only ever be called CI/github-action` section of `Makefile` as that was a conflicting thing. I could move it out of that section in `Makefile`

To answer the core of `Why should this be allowed to be run by devs and not just CI?` - Well, what this script does is simply trigger some workflows in this repo with a particular input - the release version. And not all can trigger this workflow like I already mentioned in https://github.com/vmware-tanzu/community-edition/pull/2570#discussion_r752973765 . The workflow simply runs and does a check if the release build works by installing and running some commands. A dev running this from local is not gonna affect anything, unlike https://github.com/vmware-tanzu/community-edition/blob/main/hack/release/cut-release.sh script where there's an effect - git commit, git push, PR creation, PR merge etc. So I think it's okay to not add anything like https://github.com/vmware-tanzu/community-edition/blob/main/hack/release/cut-release.sh#L36-L40 `TCE_CI_BUILD`. A run from local by a dev is not gonna negatively affect anything

Reply:
Yea, it's just a warning to users. The whole reason for putting this in the makefile is because it then becomes testable by devs.